### ISSUE TITLE
Default Termination Policy must be last

### ISSUE THREAD

**Issue by blairham** [hashicorp/terraform#3880]
When adding multiple termination policies to an autoscaling group and one of them is the 'Default', the 'Default' must be last in the list. Terraform currently does not respect that. Here is the sample code that would demonstrate this case:

variable "termination_policies" {
  default = "OldestInstance,Default"
}

resource "aws_autoscaling_group" "blue" {
...
  termination_policies = ["${split(",", var.termination_policies)}"]
...
}

**Comment by pforman** [comment]
This should be _mostly_ addressed by #2890, which was merged for 0.6.7.

The reason you're seeing that behavior is that prior to #2890 the list was implemented as a hashed set, so the order was dependent on the hash values of the items.

I fixed that so well-defined cases (like yours) will work properly.  Terraform doesn't currently allow for validating the contents of a list parameter, so if you provide something like
`termination_policies = ["Default","OldestInstance"]`
it will pass a `terraform plan` but fail on `terraform apply`
